WordPress/pattern-directory
# Add a new endpoint for pattern previews

This pulls out one change from #518, where I introduced a new endpoint for pattern previews — `wp.org/patterns/pattern/slug/view/`. Now the iframe can use that URL directly, and since the page has both the header & footer hooks in it, the theme doesn't need to do any workarounds for `__editorStyles` or generating Gutenberg styles.

Additionally, since the theme's styles are applied by WordPress rather than by loading a single theme stylesheet, the layout of the patterns looks better 🎉 

This PR also removes the `Iframe` component, since we no longer need all the special handling of injecting head & body content into an empty iframe.

Fixes #519 because we can remove the Gutenberg workaround causing that bug.

### Screenshots

|   | Before | After |
|---|--------|------|
| Grid view | ![grid-trunk](https://user-images.githubusercontent.com/541093/192007892-dafec6f3-7096-4f3e-aa1d-1badcd3aca38.png) | ![grid-after](https://user-images.githubusercontent.com/541093/192007886-d9e2d9ad-7cc4-4ae1-a0ce-14f6063d3fb1.png) |
| Single pattern | ![pattern-trunk](https://user-images.githubusercontent.com/541093/192007899-a78c6e97-74c2-43b0-9d0b-b51a5dfdd8f9.png) | ![pattern-after](https://user-images.githubusercontent.com/541093/192007896-546749b0-9a79-49ed-9b61-783bc52862cb.png) |
| Footer alignment is fixed | ![footer-trunk](https://user-images.githubusercontent.com/541093/192008169-a593825f-4b2a-4dc6-9324-706381980dc2.png) | ![footer-after](https://user-images.githubusercontent.com/541093/192008166-a1db3a16-3673-4025-b69b-5b4da08fa3c2.png) |

### How to test the changes in this Pull Request:

1. Apply the PR, build files `yarn workspaces run build`
2. `composer update` if needed to get the latest Gutenberg
3. Refresh permalinks by going to Settings > Permalinks (this will cause a change in `.htaccess`, discard it)
4. View the front page, navigate through some patterns
5. There shouldn't be any drastic visual changes
6. In the single pattern preview, you should still be able to resize the preview

Things to check: wide/full alignments & child block alignments, duotone, content spacing & colors.
